### ISSUE TITLE
Erroneous code generation when `@Value.Default` method returns `List` in some cases

### ISSUE THREAD

**Issue by grossws** [immutables/immutables#577]
Env: openjdk 8u121, immutables 2.4.4, guava 21.0

```java
@Value.Immutable
public abstract class Data {
  @Value.Parameter
  public abstract String id();

  @Value.Default
  public List<String> data() {
    return ImmutableList.of("test");
  }
}
```

fails to compile with:
```
Error:(26, 43) java: incompatible types: inferred type does not conform to upper bound(s)
    inferred: java.util.List<java.lang.String>
    upper bound(s): com.google.common.collect.ImmutableList<java.lang.String>,java.lang.Object
```

because compiled constructor contains such code:
```java
public final class ImmutableData extends Data {
  private final String id;
  private final ImmutableList<String> data;

  private ImmutableData(String id) {
    this.id = Preconditions.checkNotNull(id, "id");
    this.data = Preconditions.checkNotNull(super.data(), "data");
  }

  // other generated code
}
```

It seems that `Preconditions.checkNotNull(super.data(), "data")` should be wrapped with `ImmutableList.copyOf()` which would be almost no-op for underlaying immutable list.

**Comment by grossws** [comment]
Also using `List<? extends String>` instead of `List<String>` leads to generating slightly different (and working) code:
```java
public final class ImmutableData extends Data {
  private final String id;
  private final List<? extends String> data;

  private ImmutableData(String id) {
    this.id = Preconditions.checkNotNull(id, "id");
    this.data = Preconditions.checkNotNull(super.data(), "data");
  }

  // other generated code
}
```

**Comment by elucash** [comment]
That was an omission in a special case with a single default attribute (or multiple with `unsafeDefaultAndDerived = true` style enabled). thank you for reporting this!
the fix released in 2.4.6

**Comment by grossws** [comment]
@elucash, thanks. Works perfectly
